Updated README and CONTRIBUTING (#605)
ZupIT/horusec: README.md

    <a href="https://bestpractices.coreinfrastructure.org/projects/5146"><img src="https://bestpractices.coreinfrastructure.org/projects/5146/badge"></a>
</p>

## **Table of contents**
1. [**What is Horusec?**](#What-is-Horusec?)
2. [**Getting started**](#Getting-started)
    1. [**Requirements**](#Requirements)
    2. [**Installing Horusec**](#Installing-Horusec)
    3. [**Usage**](#Usage)
        1. [**CLI Usage**](#CLI-Usage)
        2. [**Using Docker**](#Using-Docker)
        3. [**Older versions**](#Older-versions)
        4. [**Using Horusec-Web application**](#Using-Horusec-Web-application)
        5. [**Using Visual Studio Code**](#Using-Visual-Studio-Code)
        6. [**Using the Pipeline**](#Using-the-Pipeline)
4. [**Contributing**](#Contributing)
5. [**Roadmap**](#Community)
6. [**License**](#License)
7. [**Community**](#Community)

<br>
<br>
<br>

# **What is Horusec?**

> Horusec is an open-source tool that performs a static code analysis to identify security flaws during the development process. Currently, the languages for analysis are C#, Java, Kotlin, Python, Ruby, Golang, Terraform, Javascript, Typescript, Kubernetes, PHP, C, HTML, JSON, Dart, Elixir, Shell, Nginx. 

Reviewer: ```suggestion
Horusec is an open source tool that performs a static code analysis to identify security flaws during the development process. Currently, the languages for analysis are C#, Java, Kotlin, Python, Ruby, Golang, Terraform, Javascript, Typescript, Kubernetes, PHP, C, HTML, JSON, Dart, Elixir, Shell, Nginx. 
```

> The tool has options to search for key leaks and security flaws in all your project's files, as well as in Git history. Horus can be used by the developer through the CLI and by the DevSecOps team on CI /CD mats. 

Reviewer: ```suggestion
The tool has options to search for key leaks and security flaws in all your project's files, as well as in Git history. Horusec can be used by the developer through the CLI and by the DevSecOps team on CI /CD mats. 
```

Check out our [**Documentation**](https://horusec.io/docs/overview/), you will see the complete list of tools and languages Horusec performs analysis.

<p align="center" margin="20 0"><img src="assets/horusec-complete-architecture.png" alt="architecture" width="100%" style="max-width:100%;"/></p>

### **See an Output example:**

<img src="assets/usage_horusec.gif" alt="usage_gif" width="100%" style="max-width:100%;"/>

# **Getting started**

## **Requirements**

- Docker

> To run Horusec with all the tools we use, you need Docker installed in your machine. 

Reviewer: ```suggestion
You need Docker installed in your machine in order to run Horusec with all the tools we use.
```

curl "https://github.com/ZupIT/horusec/releases/latest/download/horusec_win_x64.exe" -o "./horusec.exe" && ./horusec.exe version
```

- You can find all binaries with versions in our [**releases page**](https://github.com/ZupIT/horusec/releases).

- For more details on how to install, check out the [**documentation**](https://horusec.io/docs/cli/installation) 

#### **Check the installation**
```bash
horusec version
```

## **Usage**
### **CLI Usage**
To use horusec-cli and check the application's vulnerabilities, use the following command:
```bash
horusec start -p .
```

> {{% alert color="warning" %}}
> When horusec starts an analysis, it creates a folder called **`.horusec`**. This folder is the basis for not changing your code. We recommend you to add the line **`.horusec`** into your **`.gitignore`** file so that this folder does not need to be sent to your git server.
> {{% /alert %}}

Reviewer 2: alert not work in github markdown 
please change to: 
```suggestion
> When horusec starts an analysis, it creates a folder called **`.horusec`**. This folder is the basis for not changing your code. We recommend you to add the line **`.horusec`** into your **`.gitignore`** file so that this folder does not need to be sent to your git server.
```

curl "https://horusec.io/bin/latest/win_x64/horusec.exe" -o "./horusec.exe" && ./horusec.exe version
```

- The older binaries can be found at this endpoint, including the latest version of v1 **`v1.10.3`**. 
- As of v2, binaries will no longer be distributed by this endpoint, and you can find in the [**releases page**](https://github.com/ZupIT/horusec/releases).

### **Using Horusec-Web application**
Manage your vulnerabilities through our web interface. You can have a dashboard of metrics about your vulnerabilities, control of false positives, authorization token, update of vulnerabilities and much more.

> See more about the [**web application**](https://github.com/ZupIT/horusec-platform).

Reviewer: ```suggestion
See the [**web application**](https://github.com/ZupIT/horusec-platform) section to keep reading about it.
```

Check out the example below, it is sending an analysis to Horusec web services:
```bash
horusec start -p <PATH_TO_YOUR_PROJECT> -a <YOUR_AUTHORIZATION_TOKEN>
```

Check out [**the tutorial on how to create an authorization token through Horusec Manager Web Service**](https://horusec.io/docs/tutorials/how-to-create-an-authorization-token).

> **WARNING:** Our web services was moved to a new [**repository**](https://github.com/ZupIT/horusec-platform). You need to upgrade to v2, check out [**how to migrate from v1 to v2**](https://horusec.io/docs/migrate-v1-to-v2).

Reviewer: ```suggestion
**WARNING:** Our web services was moved to a [** new repository**](https://github.com/ZupIT/horusec-platform). You need to upgrade to v2, check out [**how to migrate from v1 to v2**](https://horusec.io/docs/migrate-v1-to-v2).
```

- Your analysis can be fully configurable, [**see all CLI available resources**](https://horusec.io/docs/cli/resources/#3-flags).

## **Contributing**

Feel free to use, recommend improvements, or contribute to new implementations.

Check out our [**contributing guide**](CONTRIBUTING.md) to learn about our development process, how to suggest bugfixes and improvements. 

### **Developer Certificate of Origin - DCO**

 This is a security layer for the project and for the developers. It is mandatory.
 
 There are two ways to use DCO, see them below: 
 
**1. Command line**
 Follow the steps: 
 **Step 1:** Check out your local git:

 ```
git config --global user.name “Name”
git config --global user.email “email@domain.com.br”
```
**Step 2:** When you commit, add the sigoff via `-s` flag:

```
$ git commit -s -m "This is my commit message"
```
**2. GitHub website**

**Step 1:** When the commit changes box opens, add 
```
$ git commit -m “My signed commit” Signed-off-by: username <email address>
```
Note: For this option, your e-mail must be the same in registered in GitHub. 

## **Roadmap**
We have a project [**roadmap**](ROADMAP.md), you can come contribute with us!

> Horus also have other repositories, check out some of our other projects:

Reviewer: ```suggestion
Horusec also have other repositories, check out some of our other projects:
```
